lib/schematype.js

```diff
@@ -451,7 +451,7 @@ SchemaType.prototype.validate = function (obj, message, type) {
 
       throw new Error(msg);
     }
-    this.validate(arg.validator, arg);
+    this.validate(arg.validator, arg.msg || arg, arg.type || undefined);
```

Comment:
This is causing an obvious test failure on Travis - don't think you can implement this in exactly the same way as you did in 3.8.x because we did [some internal refactoring of validators](http://thecodebarbarian.com/2014/12/19/mongoose-397).